### DISCUSSION TITLE
dependency build error: failed to add remote dependency

### DISCUSSION THREAD

**Discussion by machine3** [fluxcd/flux2#3541] [Q&A]
hi  My helm depends on a private helm, how do I configure the credentials of the private helm?

**Comment by hiddeco** [comment]
Create a `HelmRepository` with the URL of the private repository and the credentials, this will be resolved during the build of the chart from e.g. a `GitRepository` source.

**Comment by machine3** [comment]
When I create `HelmRepository `, I get an error:
`failed to fetch Helm repository index: failed to cache index to temporary file: Get "https://harbor.xxx": x509: certificate relies on legacy Common Name field, use SANs instead`
It does not support`--insecure-skip-tls-verify=true`

**Reply by hiddeco** [reply]
Instead of skipping the TLS verification, it would be better to renew the certificate as this has been deprecated since Go 1.15 (released in 2020). See e.g. https://stackoverflow.com/questions/64814173/how-do-i-use-sans-with-openssl-instead-of-common-name for more information on how to create a proper certificate.

**Reply by machine3** [reply]
I agree
but I'm not the maintainer of the private helm repository, I'm just a consumer, and I can't force them to do this for me. So it would be better if fluxcd is compatible with such scenarios. .
